.github/workflows/build.yml

```diff
@@ -23,31 +23,39 @@ jobs:
         with:
           filters: |
             source_changes:
-              - '.github/**'
-              - 'catalogs/**'
-              - 'clients/**'
-              - 'conf/**'
-              - 'dev/**'
-              - 'docs/open-api/**'
-              - 'gradle/**'
-              - 'gradle.properties'
-              - 'meta/**'
-              - 'scripts/**'
-              - 'web/**'
-              - '**/*.java'
-              - '**/*.kts'
-              - '**/*.py'
-              - '**/*.rs'
-              - '**/resources/**'
-              - '**/src/**'
-              - '!**/*.md'
-              - '!**/*.png'
-              - '!**/*.svg'
```

Comment:
The prefix `!` does not take effect, and we should use the `excluded` keyword to exclude some files.

Reply:
I have tried, and it seems that as long as `mcp-server` is not in the `excluded` list, the CI will be triggered even though `mcp-server` is also not in the `include` list when there are changes under `mcp-server` folder.

Reply:
This may make the configuraiton complicated, Take `spark-integration-test` for example,  we should exclude not only `mcp-server` but also `python-clients` etcs.

Reply:
> do we need to exclude mcp-server for most yml files? since mcp-server is not in include list

The fact is: `'**/*.py'` may include most files in `mcp-server`, so we need to exclude `mcp-server/**`.

Reply:
Fixed